controller-server/src/main/java/com/yahoo/vespa/hosted/controller/deployment/DeploymentTrigger.java

```diff
@@ -147,29 +137,33 @@ public long triggerReadyJobs() {
     }
 
     /**
-     * Triggers the given job for the given application.
+     * Attempts to trigger the given job for the given application and returns the outcome.
+     *
+     * If the build service can not find the given job, or claims it is illegal to trigger it,
+     * the project id is removed from the application owning the job, to prevent further trigger attemps.
      */
     public boolean trigger(Job job) {
-        log.info(String.format("Attempting to trigger %s for %s, deploying %s: %s", job.jobType, job.id, job.change, job.reason));
+        log.log(LogLevel.INFO, String.format("Attempting to trigger %s for %s, deploying %s: %s (platform: %s, application: %s)", job.jobType, job.id, job.change, job.reason, job.platformVersion, job.applicationVersion.id()));
 
-        BuildService.BuildJob buildJob = new BuildService.BuildJob(job.projectId, job.jobType.jobName());
-        if (buildService.trigger(buildJob)) {
+        try {
+            buildService.trigger(new BuildService.BuildJob(job.projectId, job.jobType.jobName()));
             applications().lockOrThrow(job.id, application -> applications().store(application.withJobTriggering(
                     job.jobType, new JobStatus.JobRun(-1, job.platformVersion, job.applicationVersion, job.reason, clock.instant()))));
             return true;
         }
-        // TODO jvenstad: On 404: set empty projectId (and send ticket?). Throw and catch NoSuchElementException.
-        // TODO jvensatd: On 401, 403 with crumb issues: refresh auth. Handle in client.
-        // TODO jvenstad: On 403 with missing collaborator status: send deployment issue ticket? Throw and catch IllegalArumentException.
-        log.log(LogLevel.WARNING, "Failed to trigger " + buildJob + " for " + job.id);
-        return false;
+        catch (RuntimeException e) {
+            if (e instanceof NoSuchElementException || e instanceof IllegalArgumentException)
+            applications().lockOrThrow(job.id, application -> applications().store(application.withProjectId(Optional.empty())));
```

Comment:
At least indent if you're not going to use `{}` 😞

Reply:
Wat. Yes.